### PR TITLE
Add support for comments inside a command file

### PR DESCRIPTION
**Pull request by HeadLessHUN**
syntax:
```
# whole line comment
// whole line comment
-SS OCTREE 10 //subsample cloud with octree level 10
 # this will run octree 10 rename_entities is commented out then save 
-SS OCTREE 10 '/* -RENAME_ENTITIES "rename" */' -SAVE_CLOUDS #we can use hashtags here as well
```

#1929

### CODE REVIEW

File: qCC/ccCommandLineCommands.cpp [hunk 1]
```diff
@@ -730,12 +730,31 @@ bool CommandLoadCommandFile::process(ccCommandLineInterface& cmd)
...
+			{
 				QString processedArg = processedArgs.takeFirst();
-				if (!processedArg.isEmpty()) {
-					cmd.print(QObject::tr("\t[%1] %2").arg(insertingIndex).arg(processedArg));
-					cmd.arguments().insert(insertingIndex, processedArg);
-					insertingIndex++;
+				if (!processedArg.isEmpty())
+				{
+					bool commentArgument = false;
+					QString commentStr = "[COMMENT] ";
```
Comment (dgirardeau) [inline qCC/ccCommandLineCommands.cpp:740]: I wouldn't initialize this string each time, as creating a QString object is quite expensive... Let me change this and you'll tell me if it's ok with you.

**Review by dgirardeau — CHANGES_REQUESTED** [review]

**Comment by dgirardeau** [comment]
Tell me what you think about the new version

**Comment by HeadLessHUN** [comment]
if we are concerned about this then, we could do early aborts on lines, and breaking out from the loop as soon as we got a line comment identifier. But in that case we won't be able to print them out at least in place. if you want i could prepare a version for this.

**Comment by dgirardeau** [comment]
Ok I'll let you test first.

I was indeed surprised that a 'full line' comment (which can actually be a partial line comment, as it can appear in the middle of the line - but that's ok, apart for the name) would still be processed in multiple chunks. I guess the chunks could be re-merged at this point, and displayed in one go (with the 'join' method of QStringList?). And then the loop can be ended earlier indeed. That's maybe the only change worthy of an effort.

**Review by dgirardeau — APPROVED** [review]

**Pull request merged**